x/subscription/keeper/subscription_test.go

```diff
@@ -874,3 +891,189 @@ func TestNextToMonthExpiryQuery(t *testing.T) {
 	require.Nil(t, err)
 	require.Equal(t, 0, len(res.Subscriptions))
 }
+
+func TestSubscriptionUpgrade(t *testing.T) {
+	ts := newTester(t)
+	ts.SetupAccounts(1, 0, 0) // 1 sub, 0 adm, 0 dev
+
+	_, consumer := ts.Account("sub1")
+	freePlan := ts.Plan("free")
+
+	// Add premium plan
+	upgradedPlan := common.CreateMockPlan()
+	upgradedPlan.Index = "premium"
+	upgradedPlan.Price = freePlan.Price.AddAmount(math.NewInt(100))
+	ts.AddPlan(upgradedPlan.Index, upgradedPlan)
+
+	// Buy free plan
+	_, err := ts.TxSubscriptionBuy(consumer, consumer, freePlan.Index, 1, false)
+	require.Nil(t, err)
+	// Verify subscription found inside getSubscription
+	getSubscriptionAndFailTestIfNotFound(t, ts, consumer)
+
+	// Charge CU from project so we can differentiate the old project from the new one
+	projectCuUsed := uint64(100)
+	project := getProjectAndFailTestIfNotFound(t, ts, consumer, ts.BlockHeight())
+	ts.Keepers.Projects.ChargeComputeUnitsToProject(ts.Ctx, project, ts.BlockHeight(), projectCuUsed)
```

Comment:
handle error

Reply:
Fixed in ec224d2d04095c69ecdfe54e5084fdcc6c4fd3c3